### PR TITLE
KAFKA-7461: Add tests for logical types

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
Added testing of logical types for Kafka Connect in support of KIP-145 features.
Added tests for Boolean, Time, Date and Timestamp, including the valid conversions.

The area of ISO8601 strings is a bit of a mess because the tokenizer is not compatible with
that format, and a subsequent JIRA will be needed to fix that.

A few small fixes as well as creating test cases, but they're clearly just corrections such as
using 0 to mean January (java.util.Calendar uses zero-based month numbers).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by AndrewJSchofield** [comment]
The contribution is my original work and that I license the work to the project under the project's open source license.

**Comment by AndrewJSchofield** [comment]
Build failure above looks like a test that fails intermittently for other PRs too.

org.apache.kafka.connect.runtime.rest.RestServerTest.testCORSEnabled

**Comment by samuel-hawker** [comment]
retest this please

**Review by mimaison — APPROVED** [review]
Thanks for the PR. LGTM

**Comment by AndrewJSchofield** [comment]
retest this please

File: connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java [hunk 1]
```diff
@@ -327,6 +347,88 @@ public void shouldParseStringsWithMultipleDelimiters() {
...
+
+        // java.util.Date as a Timestamp - discard the date and keep just day's milliseconds
+        t1 = Values.convertToTime(Timestamp.SCHEMA, current);
+        assertEquals(new java.util.Date(currentMillis), t1);
+
+        // ISO8601 strings - currently broken because tokenization breaks at colon
+
+        // Millis as string
+        java.util.Date t3 = Values.convertToTime(Time.SCHEMA, currentMillisL.toString());
+        assertEquals(currentMillisL, new Long(t3.getTime()));
```
Comment (ewencp) [inline connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java]: you can just compare the primitive longs, no need to do all this boxing to `Long` type

Comment (AndrewJSchofield) [reply connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java]: Ah, yes. I'd not realised there was an assertEquals for long values.

**Comment by AndrewJSchofield** [comment]
retest this please

**Comment by AndrewJSchofield** [comment]
retest this please

**Review by ewencp — APPROVED** [review]
LGTM, thanks for the improved test coverage!

**Pull request closed without merging**